### ISSUE TITLE
How to create pkl file of other dataset?

### ISSUE THREAD

**Issue by chituma110** [donghyeonk/fastText1607#5]
Thanks so much for sharing your code.

I have one question about the input dataset.
I am new with fastText and I want to test on other dataset in the provided drive link.
Can you provide me a method to convert other dataset to pkl version?

Thank you very much.

**Comment by donghyeonk** [comment]
You're welcome.

I plan to implement a code for each dataset in order of its size (DBP, Yelp P., ...). 

Have a nice day.

**Comment by chituma110** [comment]
Thank you for your reply. Can you tell me when will you release the related code? I have reproduced the fastText on DBP and Sogou. I got the same result on DBP, but the result on Sogou is bad.

**Comment by chituma110** [comment]
Can you show me some tips of Hashing trick and Hierarchical softmax ?

**Comment by donghyeonk** [comment]
1. You can use FeatureHasher of scikit-learn.

https://scikit-learn.org/stable/modules/feature_extraction.html#implementation-details

2. For Hierarchical softmax, refer to the following link or consider to use AdaptiveLogSoftmaxWithLoss of PyTorch.
https://talbaumel.github.io/blog/softmax/#The-Hierarchical-Softmax-Function


> Can you show me some tips of Hashing trick and Hierarchical softmax ?

**Comment by donghyeonk** [comment]
> Thank you for your reply. Can you tell me when will you release the related code? I have reproduced the fastText on DBP and Sogou. I got the same result on DBP, but the result on Sogou is bad.

It seems possible from one day to a week.

**Comment by donghyeonk** [comment]
Hi. I recently completed the implementation of fastText and found that its performance on Sogou is close to that of the original.
